jib-cli/src/test/java/com/google/cloud/tools/jib/cli/buildfile/BuildFilesTest.java

```diff
@@ -0,0 +1,109 @@
+import com.google.cloud.tools.jib.api.buildplan.AbsoluteUnixPath;
+import com.google.cloud.tools.jib.api.buildplan.ContainerBuildPlan;
+import com.google.cloud.tools.jib.api.buildplan.FileEntriesLayer;
+import com.google.cloud.tools.jib.api.buildplan.ImageFormat;
+import com.google.cloud.tools.jib.api.buildplan.LayerObject;
+import com.google.cloud.tools.jib.api.buildplan.Platform;
+import com.google.cloud.tools.jib.api.buildplan.Port;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.io.Resources;
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.time.Instant;
+import org.junit.Assert;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+public class BuildFilesTest {
+
+  @Rule public TemporaryFolder tmp = new TemporaryFolder();
+
+  @Test
+  public void testToJibContainerBuilder_allProperties()
+      throws URISyntaxException, IOException, InvalidImageReferenceException {
+    URL resource = Resources.getResource("buildfiles/projects/allProperties/jib.yaml");
+    JibContainerBuilder jibContainerBuilder =
+        BuildFiles.toJibContainerBuilder(Paths.get(resource.toURI()));
+    Path projectRoot = Paths.get(resource.toURI()).getParent();
+
+    ContainerBuildPlan resolved = jibContainerBuilder.toContainerBuildPlan();
+    Assert.assertEquals("ubuntu", resolved.getBaseImage());
+    Assert.assertEquals(Instant.ofEpochMilli(2000), resolved.getCreationTime());
+    Assert.assertEquals(ImageFormat.OCI, resolved.getFormat());
+    Assert.assertEquals(
+        ImmutableSet.of(new Platform("arm", "linux"), new Platform("amd64", "darwin")),
+        resolved.getPlatforms());
+    Assert.assertEquals(ImmutableMap.of("KEY1", "v1", "KEY2", "v2"), resolved.getEnvironment());
+    Assert.assertEquals(
+        ImmutableSet.of(AbsoluteUnixPath.get("/volume1"), AbsoluteUnixPath.get("/volume2")),
+        resolved.getVolumes());
+    Assert.assertEquals(ImmutableMap.of("label1", "l1", "label2", "l2"), resolved.getLabels());
+    Assert.assertEquals(
+        ImmutableSet.of(Port.udp(123), Port.tcp(456), Port.tcp(789)), resolved.getExposedPorts());
+    Assert.assertEquals("customUser", resolved.getUser());
+    Assert.assertEquals(AbsoluteUnixPath.get("/home"), resolved.getWorkingDirectory());
+    Assert.assertEquals(ImmutableList.of("sh", "script.sh"), resolved.getEntrypoint());
+    Assert.assertEquals(ImmutableList.of("--param", "param"), resolved.getCmd());
+
+    Assert.assertEquals(1, resolved.getLayers().size());
+    FileEntriesLayer resolvedLayer = (FileEntriesLayer) resolved.getLayers().get(0);
+    Assert.assertEquals("scripts", resolvedLayer.getName());
+    Assert.assertEquals(
+        FileEntriesLayer.builder()
+            .setName("scripts")
```

Comment:
Probably unnecessary if we are only getting entries via `getEntries()`.

Reply:
oh right, must've been leftover from when I was trying to compare FileEntriesLayer objects directly